### PR TITLE
Call ActionListener.onResponse exactly once

### PR DESCRIPTION
**Pull request by jkakavas**
Under specific circumstances we would call onResponse twice, which
led to unexpected behavior.

### CODE REVIEW

File: x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealm.java [hunk 1]
```diff
@@ -129,8 +129,9 @@ protected void doLookupUser(String username, ActionListener<User> listener) {
         if (realmEnabled == false) {
             if (anonymousEnabled && AnonymousUser.isAnonymousUsername(username, config.settings())) {
                 listener.onResponse(anonymousUser);
+            } else {
+                listener.onResponse(null);
```
Comment (jaymode) [inline x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealm.java:133]: can you add a test to ensure the listener is only called once?

Comment (jkakavas) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealm.java:133]: yes, good call, will do

Comment (tvernum) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealm.java:133]: I recently had to do exactly that, if you're looking for a possible approach, look here: https://github.com/elastic/elasticsearch/pull/60497

Comment (jkakavas) [reply x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealm.java:133]: added in ad62802

**Review by jaymode — APPROVED** [review]
LGTM

**Review by tvernum — APPROVED** [review]

**Pull request merged**